migration-0.63/main.go

```diff
@@ -0,0 +1,196 @@
+			fmt.Sprintf("INTERVAL %v SECOND DELETE", delTTL)
+		qRes := "ALTER TABLE signoz_traces.traces_v3_resource ON CLUSTER " + cluster + " MODIFY TTL toDateTime(seen_at_ts_bucket_start) + toIntervalSecond(1800) + " +
+			fmt.Sprintf("INTERVAL %v SECOND DELETE", delTTL)
+
+		if moveTTL != -1 {
+			q += fmt.Sprintf(", toDateTime(timestamp) + INTERVAL %v SECOND TO VOLUME '%s'", moveTTL, coldVolume)
+			qRes += fmt.Sprintf(", toDateTime(seen_at_ts_bucket_start) + toIntervalSecond(1800) + INTERVAL %v SECOND TO VOLUME '%s'", moveTTL, coldVolume)
+
+			for _, table := range []string{"signoz_index_v3", "traces_v3_resource"} {
+				policyReq := fmt.Sprintf("ALTER TABLE signoz_traces.%s ON CLUSTER %s MODIFY SETTING storage_policy='tiered'", table, cluster)
+				err := conn.Exec(context.Background(), policyReq)
+				if err != nil {
+					zap.S().Error(fmt.Errorf("error while setting storage policy. Err=%v", err))
+					return err
+				}
+			}
+		}
+
+		q += " SETTINGS materialize_ttl_after_modify = 0"
+		qRes += " SETTINGS materialize_ttl_after_modify = 0"
+		qS += " SETTINGS materialize_ttl_after_modify = 0"
+
+		err := conn.Exec(context.Background(), q)
+		if err != nil {
+			zap.S().Error(fmt.Errorf("error while updating ttl for table. Err=%v", err))
+			return err
+		}
+		err = conn.Exec(context.Background(), qRes)
+		if err != nil {
+			zap.S().Error(fmt.Errorf("error while updating ttl for table. Err=%v", err))
+			return err
+		}
+		err = conn.Exec(context.Background(), qS)
+		if err != nil {
+			zap.S().Error(fmt.Errorf("error while updating ttl for summary table. Err=%v", err))
+			return err
+		}
+	}
+	return nil
+}
+
+func main() {
+	logger, _ := zap.NewDevelopment()
+	defer logger.Sync()
+	undo := zap.ReplaceGlobals(logger)
+	defer undo()
+
+	start := time.Now()
+	hostFlag := flag.String("host", "127.0.0.1", "clickhouse host")
+	portFlag := flag.String("port", "9000", "clickhouse port")
+	userNameFlag := flag.String("userName", "default", "clickhouse username")
+	passwordFlag := flag.String("password", "", "clickhouse password")
+	cluster := flag.String("cluster", "cluster", "clickhouse cluster")
+	flag.Parse()
+	zap.S().Debug(fmt.Sprintf("Params: %s %s %s", *hostFlag, *portFlag, *userNameFlag))
+
+	conn, err := connect(*hostFlag, *portFlag, *userNameFlag, *passwordFlag)
+	if err != nil {
+		zap.S().Fatal("Error while connecting to clickhouse", zap.Error(err))
+	}
+
+	// get the ttl of the old table
+	delOld, moveOld, oldColdStorageName, err := GetTTL(conn, "signoz_index_v2")
+	if err != nil {
+		zap.S().Error(err.Error())
+	}
+	fmt.Println(delOld, moveOld, oldColdStorageName)
+
+	// get the ttl of the current table
+	delCurr, moveCurr, _, err := GetTTL(conn, "signoz_index_v3")
+	if err != nil {
+		zap.S().Error(err.Error())
+	}
+
+	fmt.Println(delCurr, moveCurr)
+
+	// if current table cold storage is configured don't change it.
+	if moveCurr != -1 {
+		zap.S().Info("Since cold storage is already set not changing it")
+	}
```

Comment:
Is this supposed to return or something here?

Reply:
correct, have added a proper message and exited the migration.